tests/unit/admin/test-class-wc-rest-payments-orders-controller.php

```diff
@@ -534,6 +534,358 @@ public function test_capture_terminal_payment_not_found() {
+				'id'     => $this->mock_intent_id,
+			],
+			$response_data
+		);
+
+		$result_order = wc_get_order( $order->get_id() );
+		$this->assertSame( 'woocommerce_payments', $result_order->get_payment_method() );
+		$this->assertSame( 'WooCommerce Payments', $result_order->get_payment_method_title() );
+		$this->assertSame( 'processing', $result_order->get_status() );
+	}
+
+	public function test_capture_authorization_succeeded_intent() {
+		$order = $this->create_mock_order();
+		$order->set_payment_method( WC_Payment_Gateway_WCPay::GATEWAY_ID );
+		$order->set_payment_method_title( 'WooCommerce Payments' );
+		$order->save();
+
+		$mock_intent = WC_Helper_Intention::create_intention(
+			[
+				'metadata' => [
+					'order_id' => $order->get_id(),
+				],
+			]
+		);
+
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'get_intent' )
+			->willReturn( $mock_intent );
+
+		$this->mock_gateway
+			->expects( $this->once() )
+			->method( 'capture_charge' )
+			->with( $this->isInstanceOf( WC_Order::class ) )
+			->willReturn(
+				[
+					'status'    => 'failed',
+					'message'   => 'Test error',
+					'http_code' => 502,
+				]
+			);
+
+		$request = new WP_REST_Request( 'POST' );
+		$request->set_body_params(
+			[
+				'order_id'          => $order->get_id(),
+				'payment_intent_id' => $this->mock_intent_id,
+			]
+		);
+
+		$response = $this->controller->capture_authorization( $request );
+
+		$this->assertInstanceOf( 'WP_Error', $response );
+		$data = $response->get_error_data();
+		$this->assertArrayHasKey( 'status', $data );
+		$this->assertSame( 502, $data['status'] );
+	}
+
+	public function test_capture_authorization_intent_non_capturable() {
+		$order = $this->create_mock_order();
+		$order->set_payment_method( WC_Payment_Gateway_WCPay::GATEWAY_ID );
+		$order->set_payment_method_title( 'WooCommerce Payments' );
+		$order->save();
+
+		$mock_intent = WC_Helper_Intention::create_intention( [ 'status' => 'requires_payment_method' ] );
+
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'get_intent' )
+			->willReturn( $mock_intent );
+
+		$this->mock_gateway
+			->expects( $this->never() )
+			->method( 'capture_charge' );
+
+		$this->mock_gateway
+			->expects( $this->never() )
+			->method( 'attach_intent_info_to_order' );
+
+		$this->mock_gateway
+			->expects( $this->never() )
+			->method( 'update_order_status_from_intent' );
+
+		$request = new WP_REST_Request( 'POST' );
+		$request->set_body_params(
+			[
+				'order_id'          => $order->get_id(),
+				'payment_intent_id' => $this->mock_intent_id,
+			]
+		);
+
+		$response = $this->controller->capture_authorization( $request );
+
+		$this->assertInstanceOf( 'WP_Error', $response );
+		$data = $response->get_error_data();
+		$this->assertArrayHasKey( 'status', $data );
+		$this->assertSame( 409, $data['status'] );
+	}
+
+	public function test_capture_authorization_succeeded_payment_intent_missing_order_id() {
```

Comment:
I find the name of this test confusing. `succeeded` and `missing_order_id` sound like contradictory statements.

Reply:
Indeeed, fixed here [518763f](https://github.com/Automattic/woocommerce-payments/pull/4957/commits/518763f196d41ab676cbcedbe111bf27b049d5f6) thanks!